firebase-firestore/src/main/java/com/google/firebase/firestore/local/IndexManager.java

```diff
@@ -73,7 +73,7 @@ enum IndexType {
    * <p>Values for this index are persisted asynchronously. The index will only be used for query
    * execution once values are persisted.
    */
-  void addFieldIndex(FieldIndex index);
+  void addFieldIndex(@Nullable FieldIndex index);
```

Comment:
Drive-by comment: IMO the `FieldIndex` parameter of `addFieldIndex()` should NOT be nullable. Since a null field index is nonsensical it should throw an exception in this case. Otherwise, the code could be silently ignoring bugs where null is being specified incorrectly. Instead, the caller should assume the responsibility to _not_ call the function if its field index is null.

Reply:
Agree, please address @cherylEnkidu